src/constants/types/PrincipalTypes.js

```diff
@@ -6,7 +6,8 @@
 const PrincipalTypes = {
   ORGANIZATION: 'ORGANIZATION',
   ROLE: 'ROLE',
-  USER: 'USER'
+  USER: 'USER',
+  APP: 'APP'
```

Comment:
The convention I've been following is to sort these in alphabetical order. Please move `APP` to the top.